src/main/java/org/elasticsearch/river/kafka/KafkaWorker.java

```diff
@@ -100,22 +97,10 @@ private void consumeMessagesAndAddToBulkProcessor(final KafkaStream stream) {
                 final MessageAndMetadata messageAndMetadata = consumerIterator.next();
                 logMessage(messageAndMetadata);
 
-                messageSet.add(messageAndMetadata);
```

Comment:
@schonfeld not really. There was a problem with adding a single entry to Bulk Processor while ago, and it would not put any message into Elasticsearch. It works only when we add the bulk number of messages that need to be flushed into elasticsearch at once. 
Also, in the finally block we add the remaining messages. E.g. flush interval has not reached, and Elasticsearch is stopped, then all those messages that have not been processed and sent to EL, will be done in finally block.